src/eggCookingHelper.tsx

```diff
@@ -0,0 +1,255 @@
+import { useEffect, useState } from "react";
+
+function createAudio() {
+    const audio = new Audio("/alert.mp3");
+    audio.loop = true;
+    return audio;
+}
+
+export default function EggCookingHelper() {
+    type EggSize = "S" | "M" | "L";
+    type BoilingLevel = "Soft" | "Medium" | "Hard";
+
+    const [eggSize, setEggSize] = useState<EggSize>("M");
+    const [boilingLevel, setBoilingLevel] = useState<BoilingLevel>("Medium");
+    const [remainingTimerSeconds, setRemainingTimerSeconds] = useState(0);
+    const [intervalID, setIntervalID] = useState<number>();
+    const [alertActive, setAlertActive] = useState(false);
+    const [audio] = useState(createAudio());
+
+    const cookingTimes = new Map<EggSize, Map<BoilingLevel, number>>();
+    const minutes = Math.floor(remainingTimerSeconds / 60);
+    const seconds = remainingTimerSeconds % 60;
+
+    cookingTimes.set(
+        "S",
+        new Map<BoilingLevel, number>()
+            .set("Soft", 3)
+            .set("Medium", 5)
+            .set("Hard", 7)
+    );
+
+    cookingTimes.set(
+        "M",
+        new Map<BoilingLevel, number>()
+            .set("Soft", 4)
+            .set("Medium", 6)
+            .set("Hard", 8)
+    );
+
+    cookingTimes.set(
+        "L",
+        new Map<BoilingLevel, number>()
+            .set("Soft", 5)
+            .set("Medium", 7)
+            .set("Hard", 9)
+    );
+
+    function calcBoilTime(eggSize: EggSize, boilingLevel: BoilingLevel) {
```

Comment:
Another performance optimization. Defining functions directly inside of the render cycle is usually a bad idea. You want to "cache" that function so it won't be redefined on every render.

A way to to this is the `useMemo()` hook. Feel free to look it up in the React docs. It works quite similar to the `useEffect()` hook.